File: jsimple-io/src/main/java/jsimple/io/PathNotFoundException.java
package jsimple.io;

/**
 * This is a platform independent class used to represent file not found exceptions, taking the place of
 * java.io.FileNotFoundException in JSimple based code.
 *
 * @author Bret Johnson
 * @since 10/9/12 12:15 PM
 */
public class PathNotFoundException extends IOException {
    public PathNotFoundException(Throwable cause) {
        super(cause);
    }

    public PathNotFoundException(String message) {
        super(message);
    }

    public PathNotFoundException(String message, Object arg1) {
        super(message, arg1);
    }

    public PathNotFoundException(String message, Object arg1, Object arg2) {
        super(message, arg1, arg2);
    }
}

File: jsimple-io/c#/jsimple/io/PathNotFoundException.cs
using System;

namespace jsimple.io {

    /// <summary>
    /// This is a platform independent class used to represent file not found exceptions, taking the place of
    /// java.io.FileNotFoundException in JSimple based code.
    /// 
    /// @author Bret Johnson
    /// @since 10/9/12 12:15 PM
    /// </summary>
    public class PathNotFoundException : IOException {
        public PathNotFoundException(Exception cause) : base(cause) {
        }

        public PathNotFoundException(string message) : base(message) {
        }

        public PathNotFoundException(string message, object arg1) : base(message, arg1) {
        }

        public PathNotFoundException(string message, object arg1, object arg2) : base(message, arg1, arg2) {
        }
    }

}